source/lib/types.ts

```diff
@@ -19,11 +19,28 @@ export type MyContext =
 	& ContextFlavour;
 
 export type Session = {
-	adminBroadcast?: number; // Message ID
-	adminuserquicklook?: number; // User ID
+	/** Message ID */
+	adminBroadcast?: number;
+	/** User ID */
+	adminuserquicklook?: number;
 	adminuserquicklookfilter?: string;
 	eventfilter?: string;
-	generateChangeName?: string;
+	eventAdd?: {
+		/** Path to the currently selected subdirectory on the add events screen.
+		 *
+		 * The entries of this array are the keys of the (sub)directories.
+		 */
+		eventPath?: string[];
+		/** Subdirectory item keys of the directory selected by eventPath
+		 *
+		 * This array stores the keys of subdirectories in the currently selected directory.
+		 * On the event-adding screen these keys are used to navigate into subdirectories.
+		 * Telegram callback data restricts allowed characters and length, so we store the
+		 * keys here and use the array index as the callback payload.
+		 */
+		eventDirectorySubDirectoryItems?: string[];
+	};
```

Comment:
As a thought, will this get with less typechecks when it's not involving as many undefined here? (removed the inner ?: markings)

When the object is created it can already contain (empty) arrays removing the need for these checks later on. Not sure whether it simplifies or complicates here.

```suggestion
	eventAdd?: {
		/** Path to the currently selected subdirectory on the add events screen.
		 *
		 * The entries of this array are the keys of the (sub)directories.
		 */
		eventPath: string[];
		/** Subdirectory item keys of the directory selected by eventPath
		 *
		 * This array stores the keys of subdirectories in the currently selected directory.
		 * On the event-adding screen these keys are used to navigate into subdirectories.
		 * Telegram callback data restricts allowed characters and length, so we store the
		 * keys here and use the array index as the callback payload.
		 */
		eventDirectorySubDirectoryItems: string[];
	};
```

Reply:
Ja, für `eventPath` macht es das tatsächlich einfacher. `eventDirectorySubDirectoryItems` würde ich aber so lassen, da es einen Unterschied zwischen einem leeren Ordner und einer nicht gesetzten Item-Liste gibt. Das würde glaube ich sonst mehr verwirren.